src/features/files/components/LibraryImage.tsx

```diff
@@ -1,27 +1,51 @@
-import { FC } from 'react';
+import { FC, useRef } from 'react';
 import Image from 'next/image';
 
 import TransparentGridBackground from './TransparentGridBackground';
 import { ZetkinFile } from 'utils/types/zetkin';
 
 interface LibraryImageProps {
   imageFile: ZetkinFile;
-  onLoad?: () => void;
-  onLoadingComplete?: () => void;
+  onLoad?: ({ height, width }: { height: number; width: number }) => void;
+  onLoadingComplete?: ({
+    height,
+    width,
+  }: {
+    height: number;
+    width: number;
+  }) => void;
 }
 
 const LibraryImage: FC<LibraryImageProps> = ({
   imageFile,
   onLoad,
   onLoadingComplete,
 }) => {
+  const img = useRef<HTMLImageElement>(null);
+  let dimensions = [0, 0];
+  function updateSize() {
+    if (img.current != null) {
+      dimensions = [img.current.naturalWidth, img.current.naturalHeight];
+    }
+    return true;
+  }
+
   return (
     <TransparentGridBackground>
       <Image
+        ref={img}
         alt={imageFile.original_name}
         height="400"
-        onLoad={() => onLoad && onLoad()}
-        onLoadingComplete={() => onLoadingComplete && onLoadingComplete()}
+        onLoad={() =>
+          onLoad &&
+          updateSize() &&
+          onLoad({ height: dimensions[1], width: dimensions[0] })
+        }
+        onLoadingComplete={() =>
+          onLoadingComplete &&
+          updateSize() &&
+          onLoadingComplete({ height: dimensions[1], width: dimensions[0] })
+        }
```

Comment:
I think this should be converted to a block function. The `&&` expression is confusing to me, because it doesn't clearly communicate to me that both `updateSize()` and `onLoadingComplete()` are invoked. The use of a boolean expression makes me think that `updateSize()` can have a variable return value.

The purpose of the pre-existing boolean expression was to not try to invoke `onLoadingComplete()` if it had not been defined. This additional operand obscures that intent, which IMO would be better expressed like this:

```suggestion
        onLoadingComplete={() => {
          updateSize();
          if (onLoadingComplete) {
            onLoadingComplete({ height: dimensions[1], width: dimensions[0] })
          }
        }}
```

Reply:
I agree.